git-branchless-hook/src/lib.rs

```diff
@@ -42,6 +42,41 @@ pub use lib::core::rewrite::rewrite_hooks::{
     hook_skip_upstream_applied_commit,
 };
 
+/// A component that will find the hash for a given reference transaction OID.
+/// Reference transaction OIDs may be one of:
+///
+/// * An up-to-40-hex-digit hash string
+/// * A symbolic ref, in which case, it will start with "ref:"
+///
+/// See: https://github.com/git/git/commit/a8ae923f85da6434c3faf9c39719d6d5e5c77e65
+#[derive(Debug)]
+struct TransactionReferenceFinder<'a> {
+    repo: &'a Repo,
+}
+
+impl<'a> TransactionReferenceFinder<'a> {
+    fn new(repo: &'a Repo) -> Self {
+        Self { repo }
+    }
+
+    #[instrument]
+    /// Attempt to resolve the provided string as a transaction reference OID.
+    ///
+    /// If name starts with `ref:`, and the implementation has a reference lookup
+    /// mechanism, then the OID for the provided reference name will be
+    /// returned, or a failure, if no target was found.
+    ///
+    /// Otherwise, an attempt is made to parse `name` as an up-to-40-hex-digit hash,
+    /// returning the results of that parse as a [MaybeZeroOid].
+    fn oid_for_transaction_reference(&self, name: &str) -> eyre::Result<MaybeZeroOid> {
+        match name.strip_prefix("ref:") {
+            Some(refname) => Ok(self.repo.refname_to_oid(&refname.into())?),
+
+            None => name.parse(),
```

Comment:
issue: I'm not 100% sure, but I believe this drops/doesn't include the containing function in the span-trace. You need to write `Ok(name.parse()?)` to be sure to include it. This is unfortunate language design on Rust's part.